xbmc/addons/binary-addons/AddonDll.cpp

```diff
@@ -385,6 +399,14 @@ void CAddonDll::DestroyInstance(const std::string& instanceID)
     Destroy();
 }
 
+AddonPtr CAddonDll::GetRunningInstance() const
+{
+  AddonPtr addon = CServiceBroker::GetBinaryAddonManager().GetRunningAddon(ID());
+  if (addon)
+    return addon;
+  return AddonPtr();
```

Comment:
the whole implementation of this member function can be reduced to one line: <code>return CServiceBroker::GetBinaryAddonManager().GetRunningAddon(ID());</code> ;-)